var/spack/repos/builtin/packages/py-zfit/package.py

```diff
@@ -37,6 +53,18 @@ class PyZfit(PythonPackage):
 
     # TODO: remove "build" once fixed in spack that tests need "run", not "build"
     with default_args(type=("build", "run")):
+        depends_on("py-tensorflow")
```

Comment:
Adding these two depends here makes all of the `type=("run")` redundant.

On the comment above this block, does the `type=("test", ...)` not work?

Reply:
I removed the redundant `type="run"` instances.

It is possible that the `test` dependencies are now working. I will have to check that.

Reply:
I tried digging into it and it was unclear to me what is loaded in the test environment, the logic is rather opaque to me reading in `install_test.py` and `build_environment.py` it seems like test and run deps are treated almost entirely the same.

Reply:
AFAIR, this was the issue back then. This is some spack legacy - but I think that's still the way it was, a mixup of deps. 

Thanks for taking the time to review this, it's highly appreciated!